### PR TITLE
fix(es/resolver): Skip resolving lowercase `JSXIdentifiers`

### PR DESCRIPTION
**Pull request by magic-akari**
**Related issue:**

- Closes: #9685

### CODE REVIEW

**Comment by kdy1** [comment]
IIRC, `form.input` is resolved as a `form` variable while `form` is not resolved. I'm not sure if this is the expected behavior according to the specifications.

**Comment by kdy1** [comment]
@magic-akari FYI, https://chatgpt.com/c/67218e39-87cc-8007-9af6-e396ec65c420

**Comment by magic-akari** [comment]
> @magic-akari FYI, https://chatgpt.com/c/67218e39-87cc-8007-9af6-e396ec65c420

Any conclusions?
> Unable to load conversation 67218e39-87cc-8007-9af6-e396ec65c420

**Comment by kdy1** [comment]
IMG_3467

I think the direction of the patch is correct

File: crates/swc_ecma_transforms_base/src/resolver/mod.rs [hunk 1]
```diff
@@ -920,6 +920,28 @@ impl VisitMut for Resolver<'_> {
         f.body.visit_mut_with(self);
     }
 
+    fn visit_mut_jsx_element_name(&mut self, node: &mut JSXElementName) {
+        if let JSXElementName::Ident(i) = node {
+            if i.as_ref().starts_with(|c: char| c.is_ascii_lowercase()) {
```
Comment (kdy1) [inline crates/swc_ecma_transforms_base/src/resolver/mod.rs:925]:
```suggestion
            if i.as_ref().starts_with(char::is_ascii_lowercase) {
```

Comment (magic-akari) [reply crates/swc_ecma_transforms_base/src/resolver/mod.rs]: We cannot do this because the original code implies auto dereferencing.

Comment (kdy1) [reply crates/swc_ecma_transforms_base/src/resolver/mod.rs:925]: 👍 Oh I see

**Review by kdy1 — APPROVED** [review]

**Comment by kdy1** [comment]
Thank you!

**Pull request merged**